File: pkg/pwrgrep/testdata/fixtures/c/swallowed-errors.c
#include <unistd.h>

void f(int fd) {
    // ruleid: c-swallowed-errors
    (void)close(fd);
    // ruleid: c-swallowed-errors
    if (write(fd, "x", 1) < 0) {}
    // ok: c-swallowed-errors
    if (write(fd, "y", 1) < 0) {
        _exit(1);
    }
    // ok: c-swallowed-errors
    if (fd > 2) {}
}

File: pkg/pwrgrep/testdata/fixtures/go/swallowed-errors.go
package fixture

import (
	"os"
	"strconv"
)

func f() {
	// ruleid: go-swallowed-errors
	n, _ := strconv.Atoi("1")
	// ruleid: go-swallowed-errors
	_ = os.Remove("x")
	err := os.Remove("y")
	// ruleid: go-swallowed-errors
	if err != nil {
	}
	// ok: go-swallowed-errors
	m, err := strconv.Atoi("2")
	if err != nil {
		panic(err)
	}
	_, _ = n, m
}
